circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/CollectRetained.kt

```diff
@@ -40,3 +40,26 @@ public fun <T : R, R> Flow<T>.collectAsRetainedState(
       collect { value = it }
     } else withContext(context) { collect { value = it } }
   }
+
+/**
+ * Collects values from a [Flow] provided by the [flow] lambda and represents its latest value via
+ * retained [State]. Every time there would be new value posted into the [Flow] the returned [State]
+ * will be updated causing recomposition of every [State.value] usage.
+ *
+ * This function safely collects a Flow by retaining the [inputs] across configuration changes,
+ * while the Flow itself is only remembered for the composition lifetime. This prevents memory leaks
+ * from retaining the Flow instance.
+ *
+ * @param inputs A set of inputs such that, when any of them have changed, will cause the state to
+ *   reset and [flow] to be rerun
+ * @param initial the initial value for the state
+ * @param context [CoroutineContext] to use for collecting.
+ * @param flow a lambda that provides the [Flow] to collect from
+ */
+@Composable
+public fun <T : R, R> collectAsRetainedState(
+  vararg inputs: Any?,
+  initial: R,
+  context: CoroutineContext = EmptyCoroutineContext,
+  flow: () -> Flow<T>,
+): State<R> = rememberRetained(*inputs) { flow() }.collectAsRetainedState(initial, context)
```

Comment:
Should the `flow` lambda be an input too? They're automatically memoized by the compose compiler normally but nothing's stopping a user from writing their own